src/realm/sync/client.cpp

```diff
@@ -393,12 +393,13 @@ SessionWrapperStack::~SessionWrapperStack()
 
 ClientImpl::~ClientImpl()
 {
-    bool client_destroyed_while_still_running = m_running;
-    REALM_ASSERT_RELEASE(!client_destroyed_while_still_running);
-
     // Since no other thread is allowed to be accessing this client or any of
     // its subobjects at this time, no mutex locking is necessary.
 
+    // Event Loop TODO: Until the event loop can be free-running, wait for the
+    // thread to exit before tearing down the client.
+    m_socket_provider->stop(true);
```

Comment:
should this run in a timer with a timeout until the TODO is fixed?

Reply:
This should only block until the event loop thread has exited (if it is still running) - I don't think we should add a timer, since it's an error if the stop() never returns...